custom_components/heru/config_flow.py

```diff
@@ -66,8 +69,18 @@ async def _show_config_form_user(self, user_input):
 
         user_schema = {
             vol.Required(CONF_DEVICE_NAME): cv.string,
+            vol.Required(
```

Comment:
This should be added to the OptionsFlowHandler method as well to support changing the model.

Same for the port data type bellow

Reply:
Ok, I have done it. Is there some sort of reloading done automatically to read the correct unit model after the change? 

For instance, the modbus client seems to be created only in `async_setup_entry`, but this seems to be called only at the integration initialization.

Reply:
I do not think it is 100% flawless, but I have implemented it and it worked in different use cases. You will need to restart HASS to account for the change of the model.

Reply:
> Ok, I have done it. Is there some sort of reloading done automatically to read the correct unit model after the change?
> 
> For instance, the modbus client seems to be created only in `async_setup_entry`, but this seems to be called only at the integration initialization.

Hm, not sure actually. There are some unused methods in __init__.py: ```async_unload_entry```` and ```async_reload_entry``` (from the code base I based the HERU code on...). Could it be that the ```OptionsFlowHandler``` should do its own recreate?

Reply:
I will look into the issues when saving after reconfiguring in #9